src/evm/utils.ts

```diff
@@ -0,0 +1,136 @@
+import { ethers, TransactionRequest } from 'ethers';
+import { MayanForwarderShimContractABI } from './abi';
+import { addresses, EvmForwarderParams } from '@mayanfinance/swap-sdk';
+import { Network } from '@wormhole-foundation/sdk-connect';
+
+const ForwardEth = 'forwardEth';
+const ForwardERC20 = 'forwardERC20';
+
+const MayanForwarderShimContractAddress =
+  '0x87a26566dbb3bf206634c1792a96ff4989e3f56e';
+
+function createMayanForwarderShim(functionName: string) {
+  if (functionName !== ForwardEth && functionName !== ForwardERC20) {
+    throw new Error('Unsupported function name');
+  }
+
+  const contractInterface = new ethers.Interface(MayanForwarderShimContractABI);
+
+  function encodeForwardEth(forwarderData: string, payee: string, fee: bigint) {
+    return contractInterface.encodeFunctionData(ForwardEth, [
+      forwarderData,
+      { payee, fee },
+    ]);
+  }
+
+  function encodeForwardERC20(
+    forwarderData: string,
+    tokenIn: string,
+    amountIn: bigint,
+    payee: string,
+    fee: bigint,
+  ) {
+    return contractInterface.encodeFunctionData(ForwardERC20, [
+      forwarderData,
+      tokenIn,
+      amountIn,
+      { payee, fee },
+    ]);
+  }
+
+  function encodeFunctionData(
+    forwarderData: string,
+    payee: string,
+    fee: bigint,
+    tokenIn: string,
+    amountIn: bigint,
+  ) {
+    if (functionName === ForwardEth) {
```

Comment:
I think this will need to change from === to a substring match, to account for other functions like `swapAndForwardERC20` and `swapAndForwardEth`